lib/auth/tls_test.go

```diff
@@ -837,14 +838,19 @@ func (s *TLSSuite) TestRemoteUser(c *check.C) {
 
 	// User is not authorized to perform any actions
 	// as local cluster does not trust the remote cluster yet
-	_, err = remoteClient.GetDomainName()
-	c.Assert(err, check.ErrorMatches, ".*bad certificate.*")
+	_, err = remoteClient.GetDomainName(ctx)
+	fixtures.ExpectConnectionProblem(c, err)
 
 	// Establish trust, the request will still fail, there is
 	// no role mapping set up
 	err = s.server.AuthServer.Trust(ctx, remoteServer, nil)
 	c.Assert(err, check.IsNil)
-	_, err = remoteClient.GetDomainName()
+
+	// Create fresh client now trust is established
```

Comment:
When using the gRPC transport, a fresh client must be created after trust is established, or requests using the client will continue to fail. There's precedence for this in `TLSSuite.TestRemoteBuiltInRole`.